design/one-pager-pluggable-secret-stores.md

```diff
@@ -0,0 +1,269 @@
+details secrets to [External Secret Stores] as an alpha feature. With this
+initial iteration, we have focused on the _in-tree_ implementation of secret
+stores by adding [Hashicorp's Vault] support as our first External Secret Store.
+However, this approach introduces [additional dependencies] to Crossplane core
+and would not scale once the community wants to support more secret stores.
+Even further, it won't be possible to use an in-house secret store with
+Crossplane since adding support for such a store to Crossplane core wouldn't
+make sense. For this purpose, with the initial design, we set it as a [goal] to
+support switching to an out-of-tree (plugin) model but intentionally left it
+[out of scope] to avoid the upfront complexity.
+
+With the gained confidence, we can now iterate with a more scalable option by
+supporting pluggable secret stores.
+
+## Goals
+
+The goal of this document is to propose a pluggable secret store mechanism
+that enables the development and consumption of out-of-tree secret store
+plugins.
+
+This new mechanism should:
+- Allow building/using new secret stores for Crossplane without making any
+changes in Crossplane and/or Provider repositories.
+- Provide a smooth development and consumption experience. 
+- Have secure communication with the plugin process.
+
+## Proposal
+
+This document proposes introducing a plugin mechanism for communicating with
+External Secret Stores. We will follow a similar approach as [KMS plugins] used
+by Kubernetes API Server for encryption at rest. A Secret Store plugin will
+be serving a well-defined API via gRPC. For the sake of security, plugins will
+be deployed as sidecars to Crossplane and Provider pods and communication will
+be over Unix domain sockets.
+
+![secret-store-plugin.png](images/secret-store-plugin.png)
+
+### API
+
+In addition to the existing Secret Store types, `Kubernetes` and `Vault`, we
+will define a new type named as `External`. We may consider deprecating type
+`Vault` in favor of a plugin supporting Vault to remove the dependencies already
+introduced.
+
+> This new type could be named as `Plugin` as well. However, after deprecating
+> in tree `Vault` type in favor of a Vault plugin, all **External** Secret
+> Stores will be over a plugin. So, `External` would always mean a Secret Store
+> supported by a plugin and plugin will just be an implementation detail that
+> does not need to be carried up to the API level.
+
+Configuration of this type would require passing a typed reference to another
+configuration custom resource defined by the deployed secret store plugin.
+
+```yaml
+apiVersion: gcp.crossplane.io/v1alpha1
+kind: StoreConfig
+metadata:
+  name: vault
+spec:
+  type: External
+  defaultScope: crossplane-system
+  external:
+    configRef:
+      apiVersion: secrets.crossplane.io/v1alpha1
+      kind: VaultConfig
+      name: vault-internal
+---
+apiVersion: secrets.crossplane.io/v1alpha1
```

Comment:
Yes, intentional. 
It was meant to be installed by the plugin process itself during startup. However, considering RBAC, it might make more sense to install it during plugin installation via the package manager if we go that route.